src/util/normalize-pattern.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+/**
+ * Explode and normalize a pattern into it's name and range.
```

Comment:
😁 I didn't write that. https://github.com/yarnpkg/yarn/pull/4114/files#diff-846935dd1790b71f1d658d684a6c7305L160 We can still fix the grammar though (in both places).

Actually I'm not entirely sure what this function does... I just concluded after inspecting it that it didn't depend on the rest of `src/package-request.js`, so I lifted it out to save ~2MB of weight in the bundled result.